backend/apps/project_health/services.py

```
@@ -0,0 +1,216 @@
+
+
+def _parse_repo_url(repo_url: str) -> tuple[str, str]:
+    """Extract (owner, repo_name) from a GitHub URL."""
+    parsed = urlparse(repo_url.rstrip("/"))
+    parts = parsed.path.strip("/").split("/")
+    if len(parts) < 2:
+        raise ValueError(f"Cannot parse owner/repo from URL: {repo_url}")
+    return parts[0], parts[1]
+
+
+def _github_get(path: str, token: str = None) -> dict | list:
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)